### PR TITLE
Backports v0.18 PR-13769 and 13767

### PR DESCRIPTION
**Pull request by pravblockc**
Merge #13769: Mark single-argument constructors "explicit"
Merge #13767: Remove redundant assignments (dead stores)

### CODE REVIEW

**Review by UdjinM6 — APPROVED** [review]
LGTM, utACK

File: src/test/test_dash.h [hunk 1]
```diff
@@ -143,7 +143,7 @@ CBlock getBlock13b8a();
 // BOOST_CHECK_EXCEPTION predicates to check the specific validation error
 class HasReason {
 public:
-    HasReason(const std::string& reason) : m_reason(reason) {}
+    explicit HasReason(const std::string& reason) : m_reason(reason) {}
```
Comment (PastaPastaPasta) [inline src/test/test_dash.h:146]: This should be in the other commit

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]
Need to move change to other commit

File: src/test/test_dash.h (hunk 1 above)
Comment (pravblockc) [reply src/test/test_dash.h]: fixed it, Moved to right PR commit

File: src/test/test_dash.h [hunk 2]
```diff
@@ -143,7 +143,7 @@ CBlock getBlock13b8a();
 // BOOST_CHECK_EXCEPTION predicates to check the specific validation error
 class HasReason {
 public:
-    explicit HasReason(const std::string& reason) : m_reason(reason) {}
+    HasReason(const std::string& reason) : m_reason(reason) {}
```
Comment (PastaPastaPasta) [inline src/test/test_dash.h]: need to drop this removal from 6de74e958db447f641adcdcddc7871dbc57c39b2

Comment (pravblockc) [reply src/test/test_dash.h]: done

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**